In [25]:
###########import
import os,time,glob,socket,math
import numpy as np 
import os
import string
import glob
import pandas as pd

 <font color=#0000FF>1.题目：有四个数字：1、2、3、4，能组成多少个互不相同且无重复数字的三位数？各是多少？</font>
    
  程序分析：可填在百位、十位、个位的数字都是1、2、3、4。组成所有的排列后再去 掉不满足条件的排列。

In [1]:
import string
import numpy as np
num=[1,2,3,4]
#print(num)
#type(num)
rnd=np.random.choice(num,3,replace=False)
print(rnd)


[2 3 1]


<font color=#0000FF>2.题目：企业发放的奖金根据利润提成。利润(I)低于或等于10万元时，奖金可提10%；利润高于10万元，低于20万元时，低于10万元的部分按10%提成，高于10万元的部分，可提成7.5%；20万到40万之间时，高于20万元的部分，可提成5%；40万到60万之间时高于40万元的部分，可提成3%；60万到100万之间时，高于60万元的部分，可提成1.5%，高于100万元时，超过100万元的部分按1%提成，从键盘输入当月利润I，求应发放奖金总数？</FONT>

程序分析：请利用数轴来分界，定位。注意定义时需把奖金定义成长整型。

In [6]:
i = int(input('净利润:'))
arr = [1000000,600000,400000,200000,100000,0]
rat = [0.01,0.015,0.03,0.05,0.075,0.1]
r = 0
for idx in range(0,6):
    if i>arr[idx]:
        r+=(i-arr[idx])*rat[idx]
        #print (i-arr[idx])*rat[idx]
        i=arr[idx]
print (r)

净利润:120000
11500.0


<font color=#0000FF>3.题目：一个整数，它加上100后是一个完全平方数，再加上168又是一个完全平方数，请问该数是多少？
</font>

程序分析：
假设该数为 x。
1. 则：x + 100 = n2, x + 100 + 168 = m2
2. 计算等式：m2 - n2 = (m + n)(m - n) = 168
3. 设置： m + n = i，m - n = j，i * j =168，i 和 j 至少一个是偶数
4. 可得： m = (i + j) / 2， n = (i - j) / 2，i 和 j 要么都是偶数，要么都是奇数。
5. 从 3 和 4 推导可知道，i 与 j 均是大于等于 2 的偶数。
6. 由于 i * j = 168， j>=2，则 1 < i < 168 / 2 + 1。
7. 接下来将 i 的所有数字循环计算即可。

In [7]:
for i in range(1,85):
    if 168 % i == 0:
        j = 168 / i;
        if  i > j and (i + j) % 2 == 0 and (i - j) % 2 == 0 :
            m = (i + j) / 2
            n = (i - j) / 2
            x = n * n - 100
            print(x)

-99.0
21.0
261.0
1581.0


<font color=#0000FF>5.题目：输入三个整数x,y,z，请把这三个数由小到大输出。</font>

程序分析：我们想办法把最小的数放到x上，先将x与y进行比较，如果x>y则将x与y的值进行交换，然后再用x与z进行比较，如果x>z则将x与z的值进行交换，这样能使x最小。

In [10]:
#raw_input()--將所有輸入作為字串看待
l = []
for i in range(3):
    x = int(input('integer:\n'))
    l.append(x)
l.sort()
print (l)

integer:
2
integer:
3
integer:
4
[2, 3, 4]


<font color=#0000FF>6.题目：斐波那契数列。</font>

程序分析：斐波那契数列（Fibonacci sequence），又称黄金分割数列，指的是这样一个数列：0、1、1、2、3、5、8、13、21、34、……。

在数学上，费波那契数列是以递归的方法来定义：

F0 = 0     (n=0)
F1 = 1    (n=1)
Fn = F[n-1]+ F[n-2](n=>2)

In [8]:
f=[0,1]
for i in range(1,20):
    x=f[i-1]+f[i]
    f.append(x)
print(f)    

[0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765]


<font color=#0000FF>8.题目：输出 9*9 乘法口诀表</font>

In [23]:
for i in range (1,10):
    for j in range (1,i+1):
        print('{}*{}={:2d}'.format(i,j,i*j),end=" ")
    print()

1*1= 1 
2*1= 2 2*2= 4 
3*1= 3 3*2= 6 3*3= 9 
4*1= 4 4*2= 8 4*3=12 4*4=16 
5*1= 5 5*2=10 5*3=15 5*4=20 5*5=25 
6*1= 6 6*2=12 6*3=18 6*4=24 6*5=30 6*6=36 
7*1= 7 7*2=14 7*3=21 7*4=28 7*5=35 7*6=42 7*7=49 
8*1= 8 8*2=16 8*3=24 8*4=32 8*5=40 8*6=48 8*7=56 8*8=64 
9*1= 9 9*2=18 9*3=27 9*4=36 9*5=45 9*6=54 9*7=63 9*8=72 9*9=81 


<font color=#0000FF>9.题目：古典问题：有一对兔子，从出生后第3个月起每个月都生一对兔子，小兔子长到第三个月后每个月又生一对兔子，
假如兔子都不死，问每个月的兔子总数为多少？</font>

程序分析：兔子的规律为数列1,1,2,3,5,8,13,21....

In [26]:
# f1 = 1
# f2 = 1
# for i in range(1,22):
#     print '%12ld %12ld' % (f1,f2),
#     if (i % 3) == 0:
#         print ''
#     f1 = f1 + f2
#     f2 = f1 + f2

SyntaxError: invalid syntax (<ipython-input-26-18183c49b5f5>, line 4)

<font color=#0000FF>12.题目：判断101-200之间有多少个素数，并输出所有素数。</font>

程序分析：判断素数的方法：用一个数分别去除2到sqrt(这个数)，如果能被整除，则表明此数不是素数，反之是素数。 　

In [192]:
 

ls={}
for num in range(101,201):
    for i in range(2,num):
        if num%i==0:
            ans= False
            ls[num]=ans
        else:
            ans=True
            ls[num]=ans
#ls2=dict((k,v) for v,k in ls.items())            
print(ls.items())            
#print(ls.keys())
#print(ls.values())
#ls2=dict((k,v) for v,k in ls.items())
    #print(ls2)
 

dict_items([(101, True), (102, True), (103, True), (104, True), (105, True), (106, True), (107, True), (108, True), (109, True), (110, True), (111, True), (112, True), (113, True), (114, True), (115, True), (116, True), (117, True), (118, True), (119, True), (120, True), (121, True), (122, True), (123, True), (124, True), (125, True), (126, True), (127, True), (128, True), (129, True), (130, True), (131, True), (132, True), (133, True), (134, True), (135, True), (136, True), (137, True), (138, True), (139, True), (140, True), (141, True), (142, True), (143, True), (144, True), (145, True), (146, True), (147, True), (148, True), (149, True), (150, True), (151, True), (152, True), (153, True), (154, True), (155, True), (156, True), (157, True), (158, True), (159, True), (160, True), (161, True), (162, True), (163, True), (164, True), (165, True), (166, True), (167, True), (168, True), (169, True), (170, True), (171, True), (172, True), (173, True), (174, True), (175, True), (176, True), 

<font color=#0000FF>13.题目：打印出所有的"水仙花数"，所谓"水仙花数"是指一个三位数，其各位数字立方和等于该数本身。例如：153是一个"水仙花数"，因为153=1的三次方＋5的三次方＋3的三次方。</font>

程序分析：利用for循环控制100-999个数，每个数分解出个位，十位，百位。

In [203]:
for n in range(100,1000):
    i = n // 100
    j = (n // 10)%10 
    k = n % 10
    if n == i ** 3 + j ** 3 + k ** 3:
        print(n)

153
370
371
407


<font color=#0000FF>14.题目：将一个正整数分解质因数。例如：输入90,打印出90=2 * 3 * 3 * 5。</font>

程序分析：对n进行分解质因数，应先找到一个最小的质数k，然后按下述步骤完成：
* (1)如果这个质数恰等于n，则说明分解质因数的过程已经结束，打印出即可。
* (2)如果n<>k，但n能被k整除，则应打印出k的值，并用n除以k的商,作为新的正整数你n,重复执行第一步。
* (3)如果n不能被k整除，则用k+1作为k的值,重复执行第一步。

In [15]:
num=int(input("輸入數字"))
lst=[]
ans=[]
isflag=True
while isflag:
    lst=[i for i in range(2,num+1) if num%i==0]
    k=min(lst)
    if k==num:
        isflag=False
    num=int(num/k)
    ans.append(k)
#ans='*'.join(ans)
print(ans)
    #print ('{} *'.format(k),end="")

輸入數字20
[2, 2, 5]


In [235]:
#! /usr/bin/python
# import math
# number = int(input("Enter a number: "))
# list = []
# def getChildren(num):
#     print ('*'*30)
#     isZhishu = True
#     i = 2
#     square = int(math.sqrt(num))   
#     while i <= square:
#         if num % i == 0:
#             list.append(i)
#             isZhishu = False
#             getChildren(num / i)
#             i  = 1
#             break
#     i  = 1
#     if isZhishu:
#         list.append(num)
# getChildren(number)
# print (list)

Enter a number: 12
******************************
******************************
******************************
[2, 2, 3.0]


<font color=#0000FF>18.题目：求s=a+aa+aaa+aaaa+aa...a的值，其中a是一个数字。例如2+22+222+2222+22222(此时共有5个数相加)，几个数相加由键盘控制。</font>

程序分析：关键是计算出每一项的值。

In [55]:
a=2
Tn=0
layer=3
Sn=[]
for i in range(0,layer):#0,1,2,3
    Tn = Tn + a
    a = a * 10
    Sn.append(Tn)
    print(Tn)
#print(Sn)
print('總和={}'.format(sum(Sn)))
sum(Sn)

2
22
222
總和=246


246

<font color=#0000FF>19.题目：一个数如果恰好等于它的因子之和，这个数就称为"完数"。例如6=1＋2＋3.编程找出1000以内的所有完数。</font>

程序分析：请参照程序Python 练习实例14。

In [11]:
#num=6

for num in range(1,1001):
    lst=[i for i in range(1,num) if num%i==0]
    if num==sum(lst):
        print("{}={}".format(num,lst))


6=[1, 2, 3]
28=[1, 2, 4, 7, 14]
496=[1, 2, 4, 8, 16, 31, 62, 124, 248]


<font color=#0000FF> 20.题目：一球从100米高度自由落下，每次落地后反跳回原高度的一半；
再落下，求它在第10次落地时，共经过多少米？第10次反弹多高？</font>

In [16]:
tour = []
height = []
 
hei = 100.0 # 起始高度
tim = 10 # 次数
 
for i in range(1, tim + 1):
    # 从第二次开始，落地时的距离应该是反弹高度乘以2（弹到最高点再落下）
    if i == 1:
        tour.append(hei)
    else:
        tour.append(2*hei) 
    hei /= 2
    height.append(hei)
 
print('总高度：tour = {0}'.format(sum(tour)))
print('第10次反弹高度：height = {0}'.format(height[-1]))

总高度：tour = 299.609375
第10次反弹高度：height = 0.09765625


<font color=#0000FF> 21.题目：猴子吃桃问题：猴子第一天摘下若干个桃子，当即吃了一半，还不瘾，又多吃了一个第二天早上又将剩下的桃子吃掉一半，又多吃了一个。以后每天早上都吃了前一天剩下的一半零一个。到第10天早上想再吃时，见只剩下一个桃子了。求第一天共摘了多少。</font>

程序分析：采取逆向思维的方法，从后往前推断。

In [18]:
x2 = 1
for day in range(9,0,-1):
    x1 = (x2 + 1) * 2
    x2 = x1
print (x1)

1534


<font color=#0000FF>22.题目：两个乒乓球队进行比赛，各出三人。甲队为a,b,c三人，乙队为x,y,z三人。已抽签决定比赛名单。有人向队员打听比赛的名单。a说他不和x比，c说他不和x,z比，请编程序找出三队赛手的名单。</font>

In [19]:

for i in range(ord('x'),ord('z') + 1):
    for j in range(ord('x'),ord('z') + 1):
        if i != j:
            for k in range(ord('x'),ord('z') + 1):
                if (i != k) and (j != k):
                    if (i != ord('x')) and (k != ord('x')) and (k != ord('z')):
                        print ( 'order is a -- %s\t b -- %s\tc--%s' % (chr(i),chr(j),chr(k)))

order is a -- z	 b -- x	c--y


<font color=#0000FF>23.题目：打印出如下图案（菱形）:</font>

程序分析：先把图形分成两部分来看待，前四行一个规律，后三行一个规律，利用双重for循环，第一层控制行，第二层控制列。

In [36]:
for i in range(1,5):        #1,2,3,4
    print(" "*(4-i),end="")
    for j in range(1,2*i):#1,3,5,7
        print("*",end="")
    print()   
for i in range(1,4):        #1,2,3
    print(" "*i,end="")     #1,2,3
    for j in range(1,7-(2*i-1)):     #5,3,1
        print("*",end="")
    print()     

   *
  ***
 *****
*******
 *****
  ***
   *


<font color=#0000FF>24.题目：有一分数序列：2/1，3/2，5/3，8/5，13/8，21/13...求出这个数列的前20项之和。
</font>

程序分析：请抓住分子与分母的变化规律。

In [60]:
a = 2.0
b = 1.0
s = 0
for n in range(1,21):
    s += a / b
    t = a
    a = a + b
    b = t
print (s)

32.66026079864164


<font color=#0000FF>25.题目：求1+2!+3!+...+20!的和。<font>  
    

程序分析：此程序只是把累加变成了累乘。

In [61]:

n = 0
s = 0
t = 1
for n in range(1,21):
    t *= n
    s += t
print ('1! + 2! + 3! + ... + 20! = %d' % s)

1! + 2! + 3! + ... + 20! = 2561327494111820313


<font color=#0000FF>26.题目：利用递归方法求5!。</font>

程序分析：递归公式：fn=fn_1*4!

In [63]:
def fact(j):
    sum = 0
    if j == 0:
        sum = 1
    else:
        sum = j * fact(j - 1)
    return sum
 
print (fact(5))

120


<font color=#0000FF>27.题目：利用递归函数调用方式，将所输入的5个字符，以相反顺序打印出来。</font>

In [65]:
def output(s,l):
    if l==0:
        return
    print (s[l-1])
    output(s,l-1)
 
s = input('Input a string:')
l = len(s)
output(s,l)

Input a string:sssddd
d
d
d
s
s
s


<font color=#0000FF>28.题目：有5个人坐在一起，问第五个人多少岁？他说比第4个人大2岁。问第4个人岁数，他说比第3个人大2岁。问第三个人，又说比第2人大两岁。问第2个人，说比第一个人大两岁。最后问第一个人，他说是10岁。请问第五个人多大？</font>

程序分析：利用递归的方法，递归分为回推和递推两个阶段。要想知道第五个人岁数，需知道第四人的岁数，依次类推，推到第一人（10岁），再往回推。

In [69]:
def age(n):
    if n == 1: c = 10
    else: c = age(n - 1) + 2
    return c
print (age(5))

18


<font color=#0000FF>29.题目：给一个不多于5位的正整数，要求：一、求它是几位数，二、逆序打印出各位数字。</font>

程序分析：学会分解出每一位数。

In [1]:
num=int(input("數字"))

n1=num%10 #個
n2=int(num%100/10) #十
n3=int(num%1000/100)#百
n4=int(num%10000/1000)#千
n5=int(num%100000/10000)#萬

if n5!=0:
    print("5位數:",n1,n2,n3,n4,n5)   
elif n4!=0:
    print("4位數:",n1,n2,n3,n4)
elif n3!=0:
    print("3位數",n1,n2,n3)
elif n2!=0:
    print("2位數",n1,n2) 
else:
    print("1位數",n1)    

數字3098
4位數: 8 9 0 3


<font color=#0000FF>30.题目：一个5位数，判断它是不是回文数。即12321是回文数，个位与万位相同，十位与千位相同。</font>

In [11]:
num=int(input("五位數"))
n1=num%10 #個
n2=int(num%100/10) #十
n3=int(num%1000/100)#百
n4=int(num%10000/1000)#千
n5=int(num%100000/10000)#萬
if n1==n5 and n2==n4:
    print("{}是回文數".format(num))
else:
    print("{}不是回文數".format(num))


五位數34543
34543是回文數


<font color=#0000FF>32.题目：按相反的顺序输出列表的值。</font>

In [12]:
a = ['one', 'two', 'three']
for i in a[::-1]:
    print (i)

three
two
one


<font color=#0000FF>33.题目：按逗号分隔列表。</font>

In [13]:
L = [1,2,3,4,5]
s1 = ','.join(str(n) for n in L)
print (s1)

1,2,3,4,5


<font color=#0000FF>37.题目：对10个数进行排序</font>
    
程序分析：可以利用选择法，即从后9个比较过程中，选择一个最小的与第一个元素交换，下次类推，即用第二个元素与后8个进行比较，并进行交换。

In [18]:
lst=[]
for i in range(1,11):
    num=int(input("輸入數字"))
    lst.append(num)
    print(sorted(lst) )   
    

輸入數字2
[2]
輸入數字3
[2, 3]
輸入數字4
[2, 3, 4]
輸入數字66
[2, 3, 4, 66]
輸入數字78
[2, 3, 4, 66, 78]
輸入數字2
[2, 2, 3, 4, 66, 78]
輸入數字33
[2, 2, 3, 4, 33, 66, 78]
輸入數字9
[2, 2, 3, 4, 9, 33, 66, 78]
輸入數字12
[2, 2, 3, 4, 9, 12, 33, 66, 78]
輸入數字23
[2, 2, 3, 4, 9, 12, 23, 33, 66, 78]


<font color=#0000FF>44. 題目:两个 3 行 3 列的矩阵，实现其对应位置的数据相加，并返回一个新矩阵：<font>  
    
    
X = [[12,7,3],
    [4 ,5,6],
    [7 ,8,9]]

Y = [[5,8,1],
    [6,7,3],
    [4,5,9]]

In [23]:
x=[[12,7,3],[4,5,6],[7,8,9]]
y=[[5,8,1],[6,7,3],[4,5,9]]
new=[[0,0,0],[0,0,0],[0,0,0]]
# 迭代输出行
for i in range(len(x)):
   # 迭代输出列
   for j in range(len(x[0])):
        new[i][j] = x[i][j] + y[i][j]
 
print(new)


[[17, 15, 4], [10, 12, 9], [11, 13, 18]]


<font color=#0000FF>46.题目：求输入数字的平方，如果平方运算后小于 50 则退出。</font>

In [27]:
flag=True
print ('如果输入的数字小于 50，程序将停止运行。')
while flag:
    num=int(input("數字")) 
    if num*num<50:
        print(num*num)
        flag=False
    else:
        print(num*num)

如果输入的数字小于 50，程序将停止运行。
數字8
64
數字9
81
數字0
0


<font color=#0000FF>54.题目：取一个整数a从右端开始的4〜7位。</font>

程序分析：可以这样考虑： 
* (1)先使a右移4位。 
* (2)设置一个低4位全为1,其余全为0的数。可用~(~0<<4) 
* (3)将上面二者进行&运算。

In [36]:
a = int(input('input a number:\n'))
b = a >> 4
c = ~(~0 << 4)
d = b & c
print ('%o\t%o' %(a,d))

input a number:
9
11	0


<font color=#0000FF>61.题目：打印出杨辉三角形（要求打印出10行如下图）。　</font>　

In [45]:
 a = []
for i in range(10):
    a.append([])
    for j in range(10):
        a[i].append(0)
for i in range(10):
    a[i][0] = 1
    a[i][i] = 1
for i in range(2,10):
    for j in range(1,i):
        a[i][j] = a[i - 1][j-1] + a[i - 1][j]
from sys import stdout
for i in range(10):
    for j in range(i + 1):
        stdout.write(str(a[i][j]))
        stdout.write(' ')
    print()    

1 
1 1 
1 2 1 
1 3 3 1 
1 4 6 4 1 
1 5 10 10 5 1 
1 6 15 20 15 6 1 
1 7 21 35 35 21 7 1 
1 8 28 56 70 56 28 8 1 
1 9 36 84 126 126 84 36 9 1 


<font color=#0000FF>68.题目：有n个整数，使其前面各数顺序向后移m个位置，最后m个数变成最前面的m个数</font>

In [ ]:
n = int(input('整数 n 为:\n'))
m = int(input('向后移 m 个位置为:\n'))
 
def move(array,n,m):
    array_end = array[n - 1]
    for i in range(n - 1,-1,- 1):
        array[i] = array[i - 1]
    array[0] = array_end
    m -= 1
    if m > 0:move(array,n,m)
        
number = []
for i in range(n):
    number.append(int(input('输入一个数字:\n')))
print( '原始列表:',number)
 
move(number,n,m)
 
print ('移动之后:',number)

<font color=#0000FF>69.有n个人围成一圈，顺序排号。从第一个人开始报数（从1到3报数），凡报到3的人退出圈子，问最后留下的是原来第几号的那位。</font>

In [3]:

n = int(input('请输入总人数:'))
num = []
for i in range(n):
    num.append(i + 1)
 
i = 0
k = 0
m = 0
 
while m < n - 1:
    if num[i] != 0 : k += 1
    if k == 3:
        num[i] = 0
        k = 0
        m += 1
    i += 1
    if i == n : i = 0
 
i = 0
while num[i] == 0: i += 1
print (num[i])


请输入总人数:33
7


<font color=#0000FF>76.题目：编写一个函数，输入n为偶数时，调用函数求1/2+1/4+...+1/n,当输入n为奇数时，调用函数1/1+1/3+...+1/n</font>

In [11]:
def num(n):
    s=0.0   #s可能是float
    if n%2==0:
        for i in range(1,int((n/2))+1):
            a=1/(2*i)
            s=s+a
    else:
        for j in range(1,((n+1)/2)+1):
            b=1/((2*j)-1)
            s=s+b
    return (s)#縮排
n=int(input("數字"))
num(n)

數字6


0.9166666666666666

<font color=#0000FF>80.题目：海滩上有一堆桃子，五只猴子来分。第一只猴子把这堆桃子平均分为五份，多了一个，这只猴子把多的一个扔入海中，拿走了一份。第二只猴子把剩下的桃子又平均分成五份，又多了一个，它同样把多的一个扔入海中，拿走了一份，第三、第四、第五只猴子都是这样做的，问海滩上原来最少有多少个桃子？</font>

In [13]:
i = 0
j = 1
x = 0
while (i < 5) :
    x = 4 * j
    for i in range(0,5) :
        if(x%4 != 0) :
            break
        else :
            i += 1
        x = (x/4) * 5 +1
    j += 1
print(x) 

3121.0


<font color=#0000FF>81.题目：809*??=800*??+9*?? 其中??代表的两位数, 809*??为四位数，8*??的结果为两位数，9*??的结果为3位数。求??代表的两位数，及809*??后的结果。</FONT>

In [1]:
a = 809
for i in range(10,100):
    b = i * a
    if b >= 1000 and b <= 10000 and 8 * i < 100 and 9 * i >= 100:
        print (b,' = 800 * ', i, ' + 9 * ', i)

9708  = 800 *  12  + 9 *  12


<font color=#0000FF> 83.题目：求0—7所能组成的奇数个数。</font>

程序分析：

* 组成1位数是4个。

* 组成2位数是7*4个。

* 组成3位数是7*8*4个。

* 组成4位数是7*8*8*4个。

In [24]:
sum = 4
s = 4
for j in range(2,9):
    print (sum)
    if j <= 2:
         s *= 7
    else:
        s *= 8
    sum += s
print ('sum = %d' % sum)

4
32
256
2048
16384
131072
1048576
sum = 8388608


<font color=#0000FF>85.题目：输入一个奇数，然后判断最少几个 9 除于该数的结果为整数。</font>

程序分析：999999 / 13 = 76923。

In [2]:
# num=13
# flag=True
# a=9
# Tn=0
# lst=[]
# while flag:
#     if  a%num==0:
#         flag=False 
#     else:    
#         Tn = Tn + a
#         a = a * 10
# print(Tn)    
    
    
    
    

KeyboardInterrupt: 

In [ ]:
zi = int(input('输入一个数字:\n'))
n1 = 1
c9 = 1
m9 = 9
sum = 9
while n1 != 0:
    if sum % zi == 0:
        n1 = 0
    else:
        m9 *= 10
        sum += m9
        c9 += 1
print ('%d 个 9 可以被 %d 整除 : %d' % (c9,zi,sum))
r = sum / zi
print ('%d / %d = %d' % (sum,zi,r))


In [2]:
m9=9
sum=0
for i in range(1,4):
    m9*=10
    sum += m9
    print(m9)
    print(sum)

90
90
900
990
9000
9990
90000
99990


<font color=#0000FF>89.题目：某个公司采用公用电话传递数据，数据是四位的整数，在传递过程中是加密的，加密规则如下：每位数字都加上5,然后用和除以10的余数代替该数字，再将第一位和第四位交换，第二位和第三位交换。</font>

In [3]:
a = int(input('输入四个数字:\n'))
aa = []
aa.append(a % 10)
aa.append(a % 100 / 10)
aa.append(a % 1000 / 100)
aa.append(a / 1000)
from sys import stdout
for i in range(4):
    aa[i] += 5
    aa[i] %= 10
for i in range(2):
    aa[i],aa[3 - i] = aa[3 - i],aa[i]
for i in range(3,-1,-1):
    stdout.write(str(aa[i]))

输入四个数字:
2345
09.58.457.345000000000001

In [ ]:
import numpy as np
print("歡迎來玩終極遊戲，請設定遊戲範圍")
a=int(input("從多少開始?"))
b=int(input("到多少結束?"))
ans=np.random.choice(range(a,b+1))
print(ans)
flag=True

while flag:
    usr = int(input("你猜多少?"))
    
    
    if usr==ans:
        flag=False
        print("恭喜你答對了,正確答案就是{}".format(ans))
    elif usr<ans:
         print("錯 介於{}到{}之間".format(usr,b))
         usr=a
    else:
        print("錯 介於{}到{}之間".format(a,usr))
        usr=b





歡迎來玩終極遊戲，請設定遊戲範圍
從多少開始?1
到多少結束?100
27
你猜多少?50
錯 介於1到50之間
你猜多少?4
錯 介於4到100之間
你猜多少?6
錯 介於6到100之間


In [1]:
from random import randint
 
#規定範圍並產生密碼
lowest = 1
highest = 100
answer = randint(lowest, highest)
 
#重複猜數字，直到猜對為止
while True:
    guess = input('密碼介於 ' + str(lowest) + '-' + str(highest) + ':\n>>')
 
    #檢查輸入的內容是否為數字
    try:
        guess = int(guess)  #把字串轉換成整數
    except ValueError:  #轉換失敗便要求重新輸入數字
        print('格式錯誤，請輸入數字\n')
        continue
 
        #檢查輸入的數字是否介於規定範圍內
    if guess <= lowest or guess >= highest:
        print('請輸入 ' + str(lowest) + '-' + str(highest) + ' 之間的整數\n')
        continue
 
        #判斷有沒有猜中密碼
    if guess == answer:
        print('答對了！')
        break   #猜對才跳脫迴圈
    elif guess < answer:
        lowest = guess
    else:
        highest = guess

密碼介於 1-100:
>>58
密碼介於 1-58:
>>45
密碼介於 45-58:
>>50
密碼介於 50-58:
>>52
密碼介於 50-52:
>>51
答對了！


In [3]:
import numpy as np
import string

In [14]:
rnd="0123"
while rnd[0]=="0":
    rnd="".join(np.random.choice(list(string.digits),4,replace=False))
print(rnd)    

7163


In [15]:
usr=''
while not usr.isdigit() :
    usr=input("輸入四位數")
print(usr)    
    

輸入四位數1234
1234


In [ ]:
rnd="0123"
while rnd[0]=="0":
    rnd="".join(np.random.choice(list(string.digits),4,replace=False))
print(rnd)    

usr=''
while not usr.isdigit() :
    usr=input("輸入四位數")
print(usr)   


A=B=O
for i in range(1,5):
    if usr[i]==rnd[i]:
        A=A+1
    elif usr[i] in rnd: 
        B=B+1
print()        
    